### ISSUE TITLE
Crashes when row contains fewer cells than header

### ISSUE THREAD

**Issue by joshafeinberg** [sproctor/compose-data-table#7]
I have a model that represents my data table, however I get a crash when I attempt to use it. This first example, where I loop through the rows, passes

```
    DataTable(
        columns = headers,
        modifier = Modifier.fillMaxWidth()
    ) {
        table.rows.forEach { row ->
            row {
                cell { Text("A") }
                cell { Text("B") }
                cell { Text("C") }
                cell { Text("D") }
            }
        }
    }
```

However, if I then loop through the cells, I get an error

```
    DataTable(
        columns = headers,
        modifier = Modifier.fillMaxWidth()
    ) {
        table.rows.forEach { row ->
            row {
                row.cells.forEach { cell ->
                    cell { Text("A") }
                }
            }
        }
    }
```

```
Exception in thread "AWT-EventQueue-0" java.lang.ArrayIndexOutOfBoundsException: Index 5 out of bounds for length 5
	at com.seanproctor.datatable.BasicDataTableKt$BasicDataTable$1.measure-3p2s80s(BasicDataTable.kt:220)
	at androidx.compose.ui.layout.MultiContentMeasurePolicyKt$createMeasurePolicy$1$1.measure-3p2s80s(MultiContentMeasurePolicy.kt:206)
	at androidx.compose.ui.node.InnerNodeCoordinator.measure-BRTryo0(InnerNodeCoordinator.kt:103)
	at androidx.compose.foundation.layout.FillModifier.measure-3p2s80s(Size.kt:658)
	at androidx.compose.ui.node.BackwardsCompatNode.measure-3p2s80s(BackwardsCompatNode.kt:323)
	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:155)
```

The model is pretty simple, its just

`Table(headers: List<DataColumn>, rows: List<Cell>)` where `Cell` is a wrapper around a string

**Comment by joshafeinberg** [comment]
To add, this happens on both material 2 and material 3 on version 0.5.0

**Comment by sproctor** [comment]
The variable naming here is very confusing. I'll try to look into it tomorrow, but it would be helpful to have a complete reproduction example

**Comment by joshafeinberg** [comment]
I've figured out the root issue but feel free to close if you don't feel this is a bug. Here is a minimal reproduction

```
    val rows = listOf(
        listOf("My", "First", "Row"),
        listOf("Their", "Second", "Row"),
        listOf(),
        listOf("Our", "Third", "Row"),
    )

    DataTable(
        columns = listOf(
            DataColumn { Text("Column A") },
            DataColumn { Text("Column B") },
            DataColumn { Text("Column C") }
        )
    ) {
        rows.forEach { row ->
            row {
                row.forEach { column ->
                    cell { Text(column) }
                }
            }
        }
    }
```

This crashes if you have a row with no cells. I do this for when I want a blank row but I was able to fix it locally by just ensuring that if a row has less items then the columns header that it fills it in

**Comment by sproctor** [comment]
Thanks for the report! I'll get a fixed version out as soon as I have some time.

**Comment by sproctor** [comment]
Sorry, I closed this by accident, but it's fixed in 0.5.1 which should be on maven central soon.

**Comment by sproctor** [comment]
I think I'm going to revert this change and throw an exception if a row is not full. It doesn't make much sense to have an empty row in a data table or a row with partial data. The goal of this library is not to be a general purpose table, but to implement the material data table spec (or at least help people implement it).
